pandas/tests/io/parser/test_parse_dates.py

```diff
@@ -1275,12 +1267,7 @@ def test_parse_date_time(all_parsers, data, kwargs, expected):
 def test_parse_date_fields(all_parsers):
     parser = all_parsers
     data = "year,month,day,a\n2001,01,10,10.\n2001,02,1,11."
-    result = parser.read_csv(
-        StringIO(data),
-        header=0,
-        parse_dates={"ymd": [0, 1, 2]},
-        date_parser=conv.parse_date_fields,
-    )
+    result = parser.read_csv(StringIO(data), header=0, parse_dates={"ymd": [0, 1, 2]},)
```

Comment:
same comment

Reply:
Here, I would also like to replace `date_parser=conv.parse_date_fields` with `date_parser=pd.to_datetime`, due to the same reasoning as above

Reply:
sure, same reasoning on tests as above.

its a tossup here on whether to change the impl and deprecate or just deprecate. I agree that we *do* want to change the implementation, but don't also want to break things that we maybe lightly testing.